src/intercom/back_end_binding.py

```diff
@@ -96,12 +98,36 @@ class InterComBackEndAnalysisPlugInsPublisher(InterComMongoInterface):
     def __init__(self, config=None, analysis_service=None):
         super().__init__(config=config)
         self.publish_available_analysis_plugins(analysis_service)
+        self.publish_analysis_plugins_for_catalog(analysis_service)
         self.client.close()
 
     def publish_available_analysis_plugins(self, analysis_service):
         available_plugin_dictionary = analysis_service.get_plugin_dict()
         overwrite_file(self.connections['analysis_plugins']['fs'], "plugin_dictonary", pickle.dumps(available_plugin_dictionary))
 
+    def publish_analysis_plugins_for_catalog(self, analysis_service):
+        available_plugin_dictionary_bw = analysis_service.get_plugin_catalog()
+        thread_info = load_plugin_conf(available_plugin_dictionary_bw)
+
+        for plugin in available_plugin_dictionary_bw:
+            try:
+                blacklist, whitelist = analysis_service._get_blacklist_and_whitelist_from_plugin(plugin)
+                available_plugin_dictionary_bw.update(
+                    {plugin : available_plugin_dictionary_bw[plugin] + (blacklist,) + (whitelist,)})
+            except:
+                available_plugin_dictionary_bw.update(
+                    {plugin: available_plugin_dictionary_bw[plugin] + ("unkown",) + ("unkown",)})
+                print("back_end_binding: error of plugin: %s could not find blacklist/whitelist" % plugin)
+
+            if plugin in thread_info:
+                available_plugin_dictionary_bw.update(
+                    {plugin: available_plugin_dictionary_bw[plugin] + (thread_info[plugin],)})
+            else:
```

Comment:
plugins default to single thread. Please use something like
```python
 threads = thread_info[plugin] if plugin in thread_info else 1
```

Reply:
moved into get_config_of_plugins

Reply:
Why then keep the else clause?

Reply:
.. and the if ..